File: leetcode/go/2419.longest-subarray-with-maximum-bitwise-and.go
func longestSubarray(nums []int) int {
    maxn := nums[0]
    for _, n := range nums {
        maxn = max(maxn, n)
    }
    count, ans := 0, 1
    for _, n := range nums {
        if n == maxn {
            count++
            ans = max(ans, count)
        } else {
            count = 0
        }
    }
    return ans
}

File: leetcode/cpp/2419.longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int maxn = nums[0];
        for (const int n: nums) {
            maxn = max(maxn, n);
        }
        int count = 0;
        int ans = 1;
        for (const int n: nums) {
            if (n == maxn) {
                count++;
                ans = max(ans, count);
            } else {
                count = 0;
            }
        }
        return ans;
    }
};

